lib/net/ssh/transport/algorithms.rb

```diff
@@ -153,6 +157,11 @@ def initialize(session, options = {})
           @algorithms = {}
           @pending = @initialized = false
           @client_packet = @server_packet = nil
+
+          # according to the spec, it MAY be reasonable to default to ssh-rsa (since it's technically deprecated now, probably not - but this is necessary)
+          # if we receive an EXT_INFO packet, we'll update this to use those algorithms
+          @supported_signature_algs = ["ssh-rsa"]
```

Comment:
Before this change, what was the default behavior when EXT_INFO packet is not sent?

Reply:
the default was to use `rsa-sha2-256` first, then `ssh-rsa`

Reply:
I would be nervous about changing that behavior outside of specific use cases where ssh-rsa is not required by one of the parties in the ssh connection, given that ssh-rsa relies on SHA-1, and is being deprecated by openssh...